platform-hub-web/src/app/kubernetes-tokens/kubernetes-tokens-form.component.js

```diff
@@ -0,0 +1,103 @@
+export const KubernetesTokensFormComponent = {
+  bindings: {
+    transition: '<'
+  },
+  template: require('./kubernetes-tokens-form.html'),
+  controller: KubernetesTokensFormController
+};
+
+function KubernetesTokensFormController($state, hubApiService, logger, $stateParams) {
+  'ngInject';
+
+  const ctrl = this;
+
+  const userId = $stateParams.userId || (ctrl.transition && ctrl.transition.params().userId);
+  const cluster = ctrl.transition && ctrl.transition.params().cluster;
+
+  ctrl.loading = true;
+  ctrl.saving = false;
+  ctrl.isNew = true;
+  ctrl.tokenData = null;
+  ctrl.user = null;
+  ctrl.clusters = null;
+
+  ctrl.createOrUpdate = createOrUpdate;
+
+  init();
+
+  function init() {
+    logger.info('kube token form init');
+    ctrl.isNew = !cluster;
+
+    loadUser();
+
+    ctrl.clusters = [
+      {id: 'production', name: 'Production cluster'},
+      {id: 'development', name: 'Development cluster'},
+      {id: 'ci', name: 'CI cluster'},
+      {id: 'ops', name: 'OPS cluster'}
+    ];
+
+    if (ctrl.isNew) {
+      ctrl.tokenData = {};
+      ctrl.loading = false;
+    } else {
+      loadToken();
+    }
+  }
+
+  function loadToken() {
+    ctrl.loading = true;
+    ctrl.tokenData = null;
+
+    hubApiService
+      .getKubernetesTokens(userId)
+      .then(tokens => {
+        ctrl.tokenData = tokens.find(t => {
+          return t.cluster === cluster;
+        });
```

Comment:
Can use [`_.find(…)`](https://lodash.com/docs/4.17.4#find) here to simplify this a little.